tasks/copy_to_glacier/test/test_handler.py

```diff
@@ -113,317 +129,56 @@ def test_task_happy_path(self):
-            {'Bucket': source_bucket_name, 'Key': source_key},
-            destination_bucket_name,
-            f"{collection_url_path}/{filename}",
-            ExtraArgs={
-                'StorageClass': 'GLACIER',
-                'MetadataDirective': 'COPY',
-                'ContentType': content_type
-            }
-        )])
 
-    def test_task_bucket_not_present(self):
-        """
-        Bucket not present in collection, should default to 'public' in output.
-        Copy still uses passed in bucket name.
-        # todo: Why is this the way it is?
-        """
-        collection_name = uuid.uuid4().__str__()
-        collection_version = uuid.uuid4().__str__()
-        destination_bucket_name = uuid.uuid4().__str__()
-        source_bucket_name = uuid.uuid4().__str__()
-        filename = 'test' + uuid.uuid4().__str__() + '.ext'
-        collection_url_path = uuid.uuid4().__str__()
-        content_type = uuid.uuid4().__str__()
-        source_key = f'file-staging/blah/{collection_name}__{collection_version}/{filename}'
-        granule_url = f"s3://{source_bucket_name}/{source_key}"
-
-        # todo: use 'side_effect' to verify args. It is safer, as current method does not deep-copy args
-        boto3.client = Mock()
-        s3_cli = boto3.client('s3')
-        s3_cli.copy = Mock(side_effect=[None])
-        s3_cli.head_object = Mock(return_value={'ContentType': content_type})
-
-        event = {
-            'input': [granule_url],
-            'config': {
-                CONFIG_COLLECTION_KEY: {
-                    COLLECTION_NAME_KEY: collection_name,
-                    COLLECTION_VERSION_KEY: collection_version,
-                    'files': [
-                        {
-                            'regex': '[a]+',
-                            'bucket': 'bad_bucket'
-                        }
-                    ],
-                    COLLECTION_URL_PATH_KEY: collection_url_path
-                },
-                CONFIG_BUCKETS_KEY: {
-                    'bad_bucket': {
-                        'name': 'not_glacier'
-                    },
-                    'glacier': {
-                        'name': destination_bucket_name
-                    }
+        head_object_calls = []
+        copy_calls = []
+        for i in range(0, len(source_bucket_names)):
+            head_object_calls.append(call(Bucket=source_bucket_names[i], Key=source_keys[i]))
+            copy_calls.append(call(
+                {'Bucket': source_bucket_names[i], 'Key': source_keys[i]},
+                destination_bucket_name,
+                source_keys[i],
+                ExtraArgs={
+                    'StorageClass': 'GLACIER',
+                    'MetadataDirective': 'COPY',
+                    'ContentType': content_type
                 }
-            }
+            ))
 
-        }
+        s3_cli.head_object.assert_has_calls(head_object_calls)
+        s3_cli.copy.assert_has_calls(copy_calls)
```

Comment:
I personally like locking down the call_count as well to make sure only the calls you want are present.

Reply:
:+1: